openapi/src/main/java/io/helidon/openapi/OpenApiFeature.java

```diff
@@ -0,0 +1,541 @@
+                                                                       fmt.toString());
+                                                            return r;
+                                                        });
+    }
+
+    private Optional<MediaType> chooseResponseMediaType(ServerRequest req) {
+        /*
+         * Response media type default is application/vnd.oai.openapi (YAML)
+         * unless otherwise specified.
+         */
+        Optional<String> queryParameterFormat = req.query()
+                .first(OPENAPI_ENDPOINT_FORMAT_QUERY_PARAMETER);
+        if (queryParameterFormat.isPresent()) {
+            String queryParameterFormatValue = queryParameterFormat.get();
+            try {
+                return Optional.of(QueryParameterRequestedFormat.chooseFormat(queryParameterFormatValue).mediaType());
+            } catch (IllegalArgumentException e) {
+                throw new IllegalArgumentException(
+                        "Query parameter 'format' had value '"
+                                + queryParameterFormatValue
+                                + "' but expected " + Arrays.toString(QueryParameterRequestedFormat.values()));
+            }
+        }
+
+        ServerRequestHeaders headersToCheck = req.headers();
+        if (headersToCheck.acceptedTypes().isEmpty()) {
+            WritableHeaders<?> writableHeaders = WritableHeaders.create(headersToCheck);
+            writableHeaders.add(Http.Header.ACCEPT, DEFAULT_RESPONSE_MEDIA_TYPE.toString());
+            headersToCheck = ServerRequestHeaders.create(writableHeaders);
+        }
+        return headersToCheck
+                .bestAccepted(preferredMediaTypeOrdering);
+    }
+
+
+    /**
+     * Behavior shared between the SE and MP OpenAPI feature builders.
+     *
+     * @param <B> specific concrete type of the builder
+     * @param <T> specific concrete type of {@link OpenApiFeature} the builder creates
+     */
+    public abstract static class Builder<B extends Builder<B, T>, T extends OpenApiFeature>
+            extends HelidonFeatureSupport.Builder<B, T> {
+
+        /**
+         * Config key for the OpenAPI section.
+         */
+        public static final String CONFIG_KEY = "openapi";
+
+        private OpenApiStaticFile staticFile;
+
+        private OpenApiUi.Builder<?, ?> uiBuilder = OpenApiUi.builder();
+
+        /**
+         * Constructor for the builder.
+         */
+        protected Builder() {
+            super(DEFAULT_CONTEXT);
+        }
+
+        /**
+         * Returns the logger for the OpenAPI feature instance.
+         *
+         * @return logger
+         */
+        protected abstract System.Logger logger();
+
+        /**
+         * Apply configuration settings to the builder.
+         *
+         * @param config the Helidon config instance
+         * @return updated builder
+         */
+        public B config(Config config) {
```

Comment:
you should file a tracking issue to redo this [config] builder to use the mechanism tlanger is putting into his upcoming pr.

Reply:
Planned as part of #5411.